hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/processors/StreamToStreamJoinP.java

```diff
@@ -413,6 +509,14 @@ public StreamToStreamJoinProcessorSupplier(
 
         }
 
+        @Override
+        public void init(@Nonnull Context context) throws Exception {
+            if (!joinInfo.isEquiJoin() && context.processingGuarantee() != ProcessingGuarantee.NONE) {
+                throw new UnsupportedOperationException(
+                        "Non-equi-join fault-tolerant stream-to-stream JOIN is not supported");
```

Comment:
why? what causes this limitation?

Reply:
Non-equi outer join is a subject for a separate PR

Reply:
outer join can be equi-join, can't it? We do not reject outer joins with snapshots here, do we do it different place?

Reply:
Sure, it can be. Added check to suppress outer join